polaris-core/src/main/java/org/apache/polaris/core/entity/CatalogEntity.java

```diff
@@ -279,6 +285,15 @@ public Builder setStorageConfigurationInfo(
       return this;
     }
 
+    /** Validate the number of allowed locations not exceeding the max value. */
+    public void validateMaxAllowedLocations(
+        Collection<String> allowedLocations, int maxAllowedLocations) {
+      if (maxAllowedLocations != -1 && allowedLocations.size() > maxAllowedLocations) {
+        throw new IllegalArgumentException(
+            "Number of allowed locations exceeds " + maxAllowedLocations);
```

Comment:
nit: why not state the actual number that failed the check? It could be helpful for users to trouble-shoot issues.

Reply:
nit: `Preconditions` supports failure message parameters, IIRC :)

Reply:
Is there a Preconditions method that maps to an IllegalArgumentException? I'm not too familiar with the exception mapping

Reply:
It looks like we can do:
```
      Preconditions.checkArgument(
          maxAllowedLocations != -1 && allowedLocations.size() > maxAllowedLocations,
          "Number of configured locations (%s) exceeds the limit of %s",
          allowedLocations.size(),
          maxAllowedLocations);
```

This looks a little awkward to me, so if it's okay with you I may keep the existing exception with String.format?

Reply:
sure, it was only a "nit" comment :)